image/validate_seed_test.go

```diff
@@ -0,0 +1,200 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package image_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/image"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+)
+
+type validateSuite struct {
+	imageSuite
+}
+
+var _ = Suite(&validateSuite{})
+
+var coreYaml = `name: core
+version: 1.0
+type: os`
+
+var snapdYaml = `name: snapd
+version: 1.0
+type: snapd`
+
+func (s *validateSuite) SetUpTest(c *C) {
+	s.imageSuite.SetUpTest(c)
+
+	err := os.MkdirAll(filepath.Join(s.root, "snaps"), 0755)
+	c.Assert(err, IsNil)
+}
+
+func (s *validateSuite) makeSnapInSeed(c *C, snapYaml string) {
+	info := infoFromSnapYaml(c, snapYaml, snap.R(1))
+
+	src := snaptest.MakeTestSnapWithFiles(c, snapYaml, nil)
+	dst := filepath.Join(s.root, fmt.Sprintf("snaps/%s_%s.snap", info.InstanceName(), info.Revision.String()))
```

Comment:
minor nitpick, couldn't the `snaps` be it's own argument to `filepath.Join`? I think it's more readable that way
```suggestion
	dst := filepath.Join(s.root, "snaps", fmt.Sprintf("%s_%s.snap", info.InstanceName(), info.Revision.String()))
```